### DISCUSSION TITLE
Export and corrupt PNG/JPG Images

### DISCUSSION THREAD

**Discussion by jafin** [KosmosisDire/obsidian-webpage-export#472] [Q&A]
Version: 1.8.01

When I export a page full of PNG or JPG images, the result is corrupt PNG/JPGs.
The size of the output PNG has changed, usually about 1.5x greater in size.  I would of expected it to be a 1:1 `copy` operation?

The top is the original png header, the below is what it creates after export, which does not render.

![image](https://github.com/user-attachments/assets/cd1f32d1-7f29-4567-897e-4b7c1ddbb9e3)

Is there some form of encoding that is applied to exported images. perhaps I can turn it off?

**Comment by KosmosisDire** [comment]
This is quite strange. What operating system are you on? If Linux it is possible something is going wrong there. I assume the images no longer work.
Also are you using the webpage export or the self contained documents export?

**Reply by jafin** [reply]
Running Windows 11 23H2 and using the web page export.  
You are correct, the images are no longer valid at all.

If it's not happening to anyone else, perhaps it is something on my system.  
At a guess I thought the exporter may be copying as UTF-8 instead of binary copy...  But again if no one else experiences it, something local is breaking.  I'll try to gleam more information and attempt from different Windows PC

**Reply by jafin** [reply]
As a quick test, I ran a test js app to copy the png using utf-8 encoding and it matches what I'm seeing in the above screenshot.  So  my guess is for some reason the export is exporting the png with utf-8 encoding not binary encoding.

```js
const fs = require('fs').promises;

async function copyFile(source, destination) {
  try {
    // Read the source file with UTF-8 encoding
    const content = await fs.readFile(source, 'utf-8');
    console.log('read file');
    // Write the content to the destination file with UTF-8 encoding
    await fs.writeFile(destination, content, 'utf-8');
    
    console.log(`File successfully copied from ${source} to ${destination}`);
  } catch (error) {
    console.error('Error copying file:', error);
  }
}

// Example usage
const sourceFile = 'test.png';
const destinationFile = 'out.png';

copyFile(sourceFile, destinationFile);
```
